server/src/test/java/org/elasticsearch/rest/RestControllerTests.java

```diff
@@ -623,6 +623,39 @@ public void testDispatchRestrictSystemIndices() {
         assertFalse(context.isSystemIndexAccessAllowed());
     }
 
+    public void testDispatchCompatibleHandler() {
+        final String mimeType = randomFrom("application/vnd.elasticsearch+json;compatible-with=7");
```

Comment:
can you use a random version ?

Reply:
actually in order to test a compatible handler, a Version.Current - 1 has to be used